src/core/connection.c

```diff
@@ -6567,6 +6660,217 @@ QuicConnParamSet(
         return QUIC_STATUS_SUCCESS;
     }
 
+    case QUIC_PARAM_CONN_ADD_LOCAL_ADDRESS: {
+
+        if (BufferLength != sizeof(QUIC_ADDR)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (QuicConnIsServer(Connection)) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        if (!Connection->State.Started) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        QUIC_CID_LIST_ENTRY* NewDestCid = QuicConnGetUnusedDestCid(Connection);
+        if (NewDestCid == NULL) {
+            Status = QUIC_STATUS_INVALID_STATE;
```

Comment:
I wonder if we can support a model where we always queue the path (somehow) even if we don't have a CID to use for it, _yet_...

Reply:
I've implemented this.
https://github.com/masa-koz/msquic/pull/3/files

Reply:
Now, we defer sending a path challange if there is no unused ConnID. So, I've removed `QUIC_PARAM_CONN_LOCAL_UNUSED_DEST_CID_COUNT`.
And I've modified `Basic/WithProbePathArgs.ProbePath` for testing wheter a path challenge will be sent when the peer sends a new ConnID.

https://github.com/microsoft/msquic/pull/4218/commits/c6f7e3d7ecbb21b742095104834889d5937effc9